Add js guidelines about ref naming (#35)
hypothesis/frontend-toolkit: docs/js-guide.md

class PDFIntegration { ... };
```

#### Ref naming

> Preact does not allow the `ref` prop to be passed verbatim on non-HTML components. Instead it

Reviewer: ```suggestion
Preact does not allow the `ref` prop to be accessed by non-HTML components. Instead it
```

_Verbatim_ refers specifically to the wording of some text not being changed, eg. when quoted or copied. In this context the `ref` prop can be set on a non-HTML component, but the component can't currently access it.

Author: Makes sense

> offers a wrapper method called `forwardRef` to extract the ref for you. Unfortunately the wrapper

Reviewer: Can you make `forwardRef` a link to the relevant page in the Preact docs: https://preactjs.com/guide/v10/switching-to-preact/#forwardref.

method is included with `preact/compat` which is a library we currently don't import nor want
to because of a number of global side effects. For this reason, we simply use a custom property
with a "Ref" suffix and pass the ref object to that property. For example, `buttonRef`, `panelRef`
or `fooBarRef` would all be valid names. It is then up to the component to pull out a the ref
passed in to that prop and attach it to the underlying `HTMLElement`.

e.g.

```js
/**
 * @typedef FooBarProps

>  * @prop {import('preact').Ref<HTMLInputElement>} fooBarRef

Reviewer: The `Ref<T>` type works in the example here. A caveat is that it is the most general possible ref type:

```js
	type RefObject<T> = { current: T | null };
	type RefCallback<T> = (instance: T | null) => void;
	type Ref<T> = RefObject<T> | RefCallback<T>;
```

If the component expects to be able to access the `current` property then it will need to be more specific (eg. by using `{ current: HTMLInputElement }` as the type.

Author: I'm unclear exactly what to use in the example. I don't believe we use the general case anywhere in our code base so should we use this an example?
